drivers/platform/x86/thinkpad-wmi.c

```diff
@@ -1090,13 +1192,28 @@ static void __init thinkpad_wmi_analyze(struct thinkpad_wmi *thinkpad)
 	 * and use it to create sysfs attributes */
 	for (i = 0; i < 0xFF; ++i) {
 		char *item = NULL;
+		int spleng = 0;
+		int num = 0;
 		char *p;
 
 		status = thinkpad_wmi_bios_setting(i, &item);
 		if (ACPI_FAILURE(status))
 			break;
-		if (!item || !*item)
+		//if (!item || !*item)
+		//	break;
+		if (!item )
 			break;
+		if (!*item )
+			continue;
+		
+		/* It is not allowed to have '/' for file name. Convert it into '\'. */
+		spleng = strlen(item);
+		for (num = 0; num < spleng; num++) {
```

Comment:
same as above, you can do `for (int i = 0`

Reply:
The module is currently built with -std=gnu89 (picked up  from kernel makefile) so trying to declare inside the for loop fails on compile.
error: 'for' loop initial declarations are only allowed in C99 or C11 mode.  Would prefer not to change the c standard used during compile for this module.